frontend/src/ui/index.tsx

```diff
@@ -126,12 +128,21 @@ export const Title: React.FC<TitleProps> = ({ title, className }) => (
     <h2 className={className} css={{ margin: "16px 0" }}>{title}</h2>
 );
 
-export const FOCUS_STYLE_INSET = {
+/**
+ * Applies focus outline with a default width of 2.5px to elements.
+ * This should always be used instead of a custom focus property to ensure
+ * consistency throughout the design.
+ * If ({ inset: true }) is declared, the focus will be on the inside
+ * of the focused element with a negative offset equal to the outline's width.
+ * It is also possible to declare an additional offset (usually 1) for
+ * elements with a similar color to the outline to make it stand out more.
+ */
+export const focusStyle = ({ inset = false, offset = 0, width = 2.5 }) => ({
```

Comment:
One could improve the arguments type here. Because specifying `inset` and `offset` at the same time makes no sense, the type should ideally reflect that. Maybe this:

```
type FocusStyleArgs = { width: number } & ({ inset: true } | { offset. number });

const focusStyle = ({ width, ...args }: FocusStyleArgs)
```

Reply:
I spent too much time on getting this to work the way I wanted. I had different working versions but here is the shortest I ended up with:
```
type FocusStyleArgs = { width?: number } & ({ inset?: true } | { offset?: number });

export const focusStyle = ({ width = 3, ...args }: FocusStyleArgs) => ({
    "&:focus-visible": {
        outline: `${width}px solid var(--accent-color)`,
        outlineOffset: `${"offset" in args ? args.offset : "inset" in args ? -width : 0}px`,
    },
} as const);
```
I'm not completely sure if that's what you had in mind though.
I think the properties should be optional, so the function can also work without passing any. I'm a little annoyed that I can't access the rest properties directly without checking their existence.
One thing to note is that with this, you're still not actually forbidden to pass both `inset` and `offset` props. An `xor` would be quite nice, but that's not featured in typescript afaik and while it would be possible to implement, I'm not sure we really want those extra lines of code. I guess we could also use a package for that instead, but perhaps it really is sufficient to let developers know via the type that they should only use one of these props.

Reply:
> One thing to note is that with this, you're still not actually forbidden to pass both `inset` and `offset` props.

I'm very surprised by this, but I confirmed just now. Weird. In either case, I probably agree with you that the type shenanigans here are not worth it. So I think reverting it to your original version is totally fine. It's certainly simpler to read.